pkg/common/common_test.go

```diff
@@ -67,3 +71,159 @@ func TestFind(t *testing.T) {
 		t.Error("should not have found anything in the slice")
 	}
 }
+
+func TestFetchEnv(t *testing.T) {
+	t.Run("when env var exists & has a value different from the default value then return the env var value", func(t *testing.T) {
+		key := "LOG_LEVEL"
+		defaultVal := "info"
+		val := "debug"
+		os.Setenv(key, val)
+		defer os.Unsetenv(key)
+
+		result := FetchEnv(key, defaultVal)
+
+		if result != val {
+			t.Errorf("Expected %v, but got %v", val, result)
+		}
+	})
+	t.Run("when env var exists but has an empty value then return the empty value", func(t *testing.T) {
+		key := "LOG_MODE"
+		defaultVal := "production"
+		val := ""
+		os.Setenv(key, val)
+		defer os.Unsetenv(key)
+
+		result := FetchEnv(key, defaultVal)
+
+		if result != val {
+			t.Errorf("Expected %v, but got %v", val, result)
+		}
+	})
+	t.Run("when env var does not exist & the default value is used then return the default value", func(t *testing.T) {
+		key := "LOG_MODE"
+		defaultVal := "production"
+
+		result := FetchEnv(key, defaultVal)
+
+		if result != defaultVal {
+			t.Errorf("Expected %v, but got %v", defaultVal, result)
+		}
+	})
+	t.Run("when a runtime error occurs during execution then return the default value", func(t *testing.T) {
```

Comment:
where is the runtime error?

Reply:
I double-checked and yes the previous test-case covers that. Unsetting the env value leads to the same state as 'env value does not exist', so you are right, it is redundant! I am going to get rid of this test-case
Thanks, @eguzki !